docs/topics/db/optimization.txt

```diff
@@ -411,4 +416,104 @@ objects to reduce the number of SQL queries. For example::
     my_band.members.add(me)
     my_band.members.add(my_friend)
 
-...where ``Bands`` and ``Artists`` have a many-to-many relationship.
+...where ``Band`` and ``Artist`` have a many-to-many relationship.
+
+If you have a list of objects, use the ``*``-operator when calling
+:meth:`~django.db.models.fields.related.RelatedManager.add`::
+
+    my_band.members.add(*new_members)
+
+Many source objects, many target objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When inserting pairs of objects into :class:`ManyToManyFields
+<django.db.models.ManyToManyField>`, use 
+:meth:`~django.db.models.query.QuerySet.bulk_create()` with multiple
+:attr:`through <django.db.models.ManyToManyField.through>` model instances 
+to reduce the number of SQL queries. For example::
+
+    PizzaToppingRelation = Pizza.toppings.through
+    PizzaToppingRelation.objects.bulk_create([
+        PizzaToppingRelation(pizza=my_pizza, topping=pepperoni),
+        PizzaToppingRelation(pizza=your_pizza, topping=pepperoni),
+        PizzaToppingRelation(pizza=your_pizza, topping=mushroom),
+    ], ignore_conflicts=True)
+
+...is preferable to::
+
+    my_pizza.toppings.add(pepperoni)
+    your_pizza.toppings.add(pepperoni, mushroom)
+
+...where ``Pizza`` and ``Topping`` have a many-to-many relationship.
+
+If you have a list of object pairs, use the ``*``-operator when calling
+:meth:`~django.db.models.query.QuerySet.bulk_create()`::
```

Comment:
Is this right?  `bulk_create()` takes a list as its first argument so I don't understand why you'd unpack it into positional arguments.

Reply:
Fixed.